[SPI checking] Use C-style symbol names in allowlists

emw-apple opened the pull request:
#### 31979809001642aa1d3290c3dd30be0be3b0f738
<pre>
[SPI checking] Use C-style symbol names in allowlists
<a href="https://bugs.webkit.org/show_bug.cgi?id=298462">https://bugs.webkit.org/show_bug.cgi?id=298462</a>
<a href="https://rdar.apple.com/159957130">rdar://159957130</a>

Reviewed by Brianna Fan.

Instead of writing the raw symbol name in allowlists, use the C
namespace rules, i.e. remove the leading underscore from the symbol
name. This has the nice benefit of making a symbol&apos;s spelling in an
allowlist match its spelling in code.

This is a cosmetic detail. Internally, webkitapipy continues to track
symbols using their raw names, so the only change is to allowlist
parsing and reporting logic.

* Source/JavaScriptCore/Configurations/AllowedSPI-legacy.toml:
* Source/JavaScriptCore/Configurations/AllowedSPI.toml:
* Source/WebCore/Configurations/AllowedSPI-legacy.toml:
* Source/WebCore/Configurations/AllowedSPI.toml:
* Source/WebKit/Configurations/AllowedSPI-legacy.toml:
* Source/WebKit/Configurations/AllowedSPI.toml:
* Source/WebKitLegacy/mac/Configurations/AllowedSPI-legacy.toml:
* Tools/Scripts/libraries/webkitapipy/webkitapipy/allow.py:
(AllowList.from_dict):
* Tools/Scripts/libraries/webkitapipy/webkitapipy/allow_unittest.py:
* Tools/Scripts/libraries/webkitapipy/webkitapipy/reporter.py:
(Reporter.demangle_name): Added. Shared function to turn a diagnostic&apos;s
  &quot;name&quot; field into something human-readable. In the future, we may
  choose to demangle further, to turn Swift and C++ symbols into
  human-readable names.
(TSVReporter.format_diagnostic):
(BuildToolReporter.format_diagnostic):
* Tools/Scripts/libraries/webkitapipy/webkitapipy/sdkdb_unittest.py:

Canonical link: <a href="https://commits.webkit.org/299714@main">https://commits.webkit.org/299714@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/723736ca3946842be4f349b88db68a1ad0f6eb04

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119680 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39372 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30024 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/125969 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71752 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d7e377c3-1f64-46b2-829d-8a8be3503d67) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121557 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40068 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47950 "Built successfully") | [⏳ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/WPE-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60195 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/88d46799-2e04-4fd9-8817-a8beee9ed435) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122630 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31987 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107312 "Passed tests") | [⏳ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/API-Tests-WPE-EWS "Waiting to run tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7fcecd88-c629-4980-9348-077118542684) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/119037 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31021 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25416 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/69611 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/111805 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101455 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25608 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128926 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/118196 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46600 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35303 "Found 1 new test failure: http/tests/cache/disk-cache/speculative-validation/cacheable-redirect.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99501 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/46965 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103504 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99344 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44773 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22791 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43164 "Hash 723736ca for PR 50377 does not build (failure)") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19071 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46462 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52168 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/146894 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/45928 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37748 "Found 1 new JSC binary failure: testapi, Found 18680 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/array_sort2.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/toLocaleString.js.default, ChakraCore.yaml/ChakraCore/test/Basics/flags.js.default, ChakraCore.yaml/ChakraCore/test/ControlFlow/forInPrimitive.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default, ChakraCore.yaml/ChakraCore/test/Function/FuncBody.bug227901.js.default ... (failure)") | | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49277 "Build is in progress. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47614 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6f8e818b021eff00a2bf215833707b0f924f4dda. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50377

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5d5309e2627d668c896c63c7645a7e1db9bd5f4a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50377

briannafan commented on Tools/Scripts/libraries/webkitapipy/webkitapipy/allow.py:
```diff
@@ -91,14 +91,19 @@ def from_dict(cls, doc: dict[str, Any]) -> AllowList:
                     receiver = sel.get('class')
                     sels.append(AllowedSPI.Selector(sel['name'],
                                                     None if receiver == '?' else receiver))
+                # Symbols use C-style namespacing (leading underscore), so that
```
In the commit message, you say `C namespace rules, i.e. remove the leading underscore`. Here, `C-style namespacing (leading underscore)` seems to imply the opposite. Is this a typo or am I misunderstanding?

emw-apple replied:
It's poorly worded, perhaps I can say "implicit leading underscore"

briannafan commented on Tools/Scripts/libraries/webkitapipy/webkitapipy/reporter.py:
```diff
@@ -25,6 +25,13 @@ def emit_diagnostic(self, diag: Diagnostic) -> bool:
+        if diag.kind == SYMBOL:
+            # FIXME: Consider using c++filt and swift-demangle in addition to
+            # C-style namespacing.
+            return diag.name.removeprefix('_')
```
`"Demangle" them to the raw form` in `allow.py` refers to adding the _ prefix to get the raw symbol name. Here, demangling seems to refer to the opposite (removing the _ prefix to get to the C-style name). Which definition of demangle is correct?

emw-apple committed:
[SPI checking] Use C-style symbol names in allowlists
https://bugs.webkit.org/show_bug.cgi?id=298462
rdar://159957130

Reviewed by Brianna Fan.

Instead of writing the raw symbol name in allowlists, use the C
namespace rules, i.e. remove the leading underscore from the symbol
name. This has the nice benefit of making a symbol's spelling in an
allowlist match its spelling in code.

This is a cosmetic detail. Internally, webkitapipy continues to track
symbols using their raw names, so the only change is to allowlist
parsing and reporting logic.

* Source/JavaScriptCore/Configurations/AllowedSPI-legacy.toml:
* Source/JavaScriptCore/Configurations/AllowedSPI.toml:
* Source/WebCore/Configurations/AllowedSPI-legacy.toml:
* Source/WebCore/Configurations/AllowedSPI.toml:
* Source/WebKit/Configurations/AllowedSPI-legacy.toml:
* Source/WebKit/Configurations/AllowedSPI.toml:
* Source/WebKitLegacy/mac/Configurations/AllowedSPI-legacy.toml:
* Tools/Scripts/libraries/webkitapipy/webkitapipy/allow.py:
(AllowList.from_dict):
* Tools/Scripts/libraries/webkitapipy/webkitapipy/allow_unittest.py:
* Tools/Scripts/libraries/webkitapipy/webkitapipy/reporter.py:
(Reporter.demangle_name): Added. Shared function to turn a diagnostic's
  "name" field into something human-readable. In the future, we may
  choose to demangle further, to turn Swift and C++ symbols into
  human-readable names.
(TSVReporter.format_diagnostic):
(BuildToolReporter.format_diagnostic):
* Tools/Scripts/libraries/webkitapipy/webkitapipy/sdkdb_unittest.py:

Canonical link: https://commits.webkit.org/299714@main
